connectors/gitlab-pr/src/main/java/com/vmware/connectors/gitlab/pr/GitlabPrController.java

```diff
@@ -0,0 +1,575 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.AsyncRestOperations;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+import rx.Observable;
+import rx.Single;
+
+import javax.validation.Valid;
+import java.time.format.DateTimeFormatter;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@RestController
+public class GitlabPrController {
+
+    private static final Logger logger = LoggerFactory.getLogger(GitlabPrController.class);
+
+    private static final String AUTH_HEADER = "x-gitlab-pr-authorization";
+    private static final String BASE_URL_HEADER = "x-gitlab-pr-base-url";
+    private static final String ROUTING_PREFIX = "x-routing-prefix";
+
+    private static final String CLOSE_REASON_PARAM_KEY = "reason";
+    private static final String COMMENT_PARAM_KEY = "message";
+    private static final String SHA_PARAM_KEY = "sha";
+
+    // Number of path segments expected in our pull request urls
+    private static final int NUM_PR_PATH_SEGMENTS = 4;
+
+    // Merge Request path name expected
+    private static final String MERGE_REQUESTS = "merge_requests";
+
+    private final boolean isEnterpriseEdition;
+    private final AsyncRestOperations rest;
+    private final CardTextAccessor cardTextAccessor;
+
+    @Autowired
+    public GitlabPrController(
+            @Value("${gitlab.connector.enterprise:false}") boolean isEnterpriseEdition,
+            AsyncRestOperations rest,
+            CardTextAccessor cardTextAccessor
+    ) {
+        this.isEnterpriseEdition = isEnterpriseEdition;
+        this.rest = rest;
+        this.cardTextAccessor = cardTextAccessor;
+    }
+
+    @PostMapping(
+            path = "/cards/requests",
+            produces = MediaType.APPLICATION_JSON_VALUE,
+            consumes = MediaType.APPLICATION_JSON_VALUE
+    )
+    public Single<ResponseEntity<Cards>> getCards(
+            @RequestHeader(AUTH_HEADER) String auth,
+            @RequestHeader(BASE_URL_HEADER) String baseUrl,
+            @RequestHeader(ROUTING_PREFIX) String routingPrefix,
+            @Valid @RequestBody CardRequest request
+    ) {
+        logger.trace("getCards called: baseUrl={}, routingPrefix={}, request={}", baseUrl, routingPrefix, request);
+
+        List<MergeRequestId> mergeRequestUrls = request.getTokens("merge_request_urls")
```

Comment:
Type is `List<MergeRequestId>` but name is `mergeRequestUrls`.

Reply:
In my defense, it started off as URLs :)